core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -163,6 +163,20 @@ sealed trait ZValidation[+W, +E, +A] { self =>
       case Success(w, a) => Success(f(w), a)
     }
 
+  /**
+   * Returns a ZValidation that if succeeds is the result of the first or if fails is
+   * the result of the second.  The second will combine the warnings and errors of the first
+   * into the warnings of the second, then append any new warnings from the second.
+   */
+  final def or[W1 >: W, E1, E2 >: E, W2, B >: A](
+    that: ZValidation[W1, E1, B]
+  )(implicit lub: Lub[E2, W1, W2]): ZValidation[W2, E1, B] =
```

Comment:
```scala
  final def or[W1 >: W, E1 >: E, A1 >: A](
    that: ZValidation[W1, E1, A1]
  )(implicit ev: E <:< W1): ZValidation[W1, E1, A1] =
```
I'm wondering whether we could go with just this :thinking: 

Not as powerful and flexible as `Lub`, but we wouldn't need `Lub`...

Reply:
Definitely agree about trying to avoid adding additional abstractions.

I'm also not sure that this is the most natural way to deal with an error in the alternative validation. It seems like there are several possible semantics:

1. Just fail with the new error. We "recovered" from the original error so it is no longer a cause of the failure.
2. Fail with both the original error and the new error, combining them with an `Associative` instance defined for `E`. This preserves the most information but arguably goes against the intent of us "recovering" from this error.
3. Push the original error into the log type. This is to some extent an intermediate between the first two but requires unifying the log and error types which is fairly constraining. It also doesn't really work at all if you are using `Validation` rather than `ZValidation` since it doesn't return a `Validation` anymore.

I would tend to think that (1) is the most intuitive, though I think you could debate between (1) and (2). You could call that `orElse`. Then I think (3) becomes something like `orElseLog` which not only recovers from the error but also adds the error to the log and has this additional constraint.

Reply:
How about we have
```scala
  final def orWith[W1 >: W, E1 >: E, A1 >: A](convert: E => W1)(that: ZValidation[W1, E1, A1]): ZValidation[W1, E1, A1] = ???
```
or
```scala
  final def orWith[W1 >: W, E1 >: E, A1 >: A](convert: Chunk[E] => Chunk[W1])(that: ZValidation[W1, E1, A1]): ZValidation[W1, E1, A1] = ???
```
?

Reply:
The thought behind keeping the original error around would be to know for example if a default was applied.
For example if an external service was unavailable during the time period of execution.  The data might still be valid, however perhaps a backup method is not as accurate and we may wish to schedule some further data process at a later time when it comes back online.

I suppose there is probably other ways to do this, such as always add warning log to the default that gets applied.

Something like
```
verifyItem or (default.log("default applied"))
```

Let me tinker around with the `orWith` version and see if that fits the use-cases I'm trying to solve.  I like that idea of the conversion as its even more general than expecting `E` and `W` to inherit from the same root object.